### PR TITLE
Silence unused lambda warning

### LINKED ISSUE #7949: Recursor: unused lambda capture in unit tests
- Program: Recursor
 - Issue type: Bug report/Feature request 

### Short description
Compiler warning when compiling `test-syncres_cc8.cc`

### Environment

 - Operating system: Arch Linux
 - Software version: git (d26856a32e7506817c291c49d8eb5a60129dd751)
 - Software source: git

### Steps to reproduce
```
autoreconf -fi
CC=clang CXX=clang++ LD=lld ./configure --enable-unit-tests
make
```

### Expected behaviour
No warnings.

### Actual behaviour
```
test-syncres_cc8.cc:701:51: warning: lambda capture 'fixedNow' is not used [-Wunused-lambda-capture]
  sr->setAsyncCallback([target,&queriesCount,keys,fixedNow](const ComboAddress& ip, const DNSName& domain, int type, bool doTCP, bool sendRDQuery, int EDNS0Level, struct timeval* now, boost::optional<Netmask>& srcmask, boost::optional<const ResolveContext&> context, LWResult* res, bool* chained) {
                                                 ~^~~~~~~~
1 warning generated.
```

### PR DESCRIPTION
**Pull request by fwSmit**
Fixes #7949
This silences the unused lambda warning by only defining it when it's used.

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [ ] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: pdns/recursordist/test-syncres_cc8.cc [hunk 1]
```diff
@@ -698,6 +698,7 @@ BOOST_AUTO_TEST_CASE(test_dnssec_rrsig_negcache_bogus_validity) {
   size_t queriesCount = 0;
   const time_t fixedNow = sr->getNow().tv_sec;
 
+  #if defined(HAVE_DNS_OVER_TLS) || defined(HAVE_DNS_OVER_HTTPS)
```
Comment (pieterlexis) [inline pdns/recursordist/test-syncres_cc8.cc]: This is really not the right way.

You should remove the unused vars from between the square brackets

Comment (fwSmit) [reply pdns/recursordist/test-syncres_cc8.cc]: I changed it. Can you check if it's good now?

Comment (fwSmit) [reply pdns/recursordist/test-syncres_cc8.cc]: Now it's updated correctly. Can you have a look at it?

Comment (omoerbeek) [reply pdns/recursordist/test-syncres_cc8.cc]: This looks good., although I do not understand why mu clang (7.0.1) does not warn about `target`.
Also: can you please squash the commits into a single one? There's nothing from your earlier versions we would like to preserve.

Comment (fwSmit) [reply pdns/recursordist/test-syncres_cc8.cc]: i tried to squash the commits, but as you can see: it didn't work too well. Could you help me with that? I'm kinda new to forks.

Comment (fwSmit) [reply pdns/recursordist/test-syncres_cc8.cc]: I'll just make another pull request. That's easier

**Pull request closed without merging**